presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -232,11 +266,13 @@ private SqlQueryScheduler(
 
         this.rootStageId = Iterables.getLast(stageExecutions).getStageExecution().getStageExecutionId().getStageId();
 
-        this.stageExecutions = stageExecutions.stream()
-                .collect(toImmutableMap(execution -> execution.getStageExecution().getStageExecutionId().getStageId(), identity()));
+        this.stageExecutions = synchronizedMap(
```

Comment:
Curious: What's the difference between synchronized map and concurrent hash map ?

Reply:
For inserts synchronized map locks the whole map.  ConcurrentHashMap only locks particular keys.  So, putAll is atomic for a synchronized map, but not for a ConcurrentHashMap